tests/test_color.py

```diff
@@ -219,3 +230,24 @@ def test_rgb_lab_rgb_roundtrip():
 def test_color_pair_style():
     pair = ColorPair(Color(220, 220, 220), Color(10, 20, 30))
     assert str(pair.style) == "#dcdcdc on #0a141e"
+
+
+@pytest.mark.parametrize(
+    "expression,expected_color,expected_is_auto",
+    (
+        ["red", Color(255, 0, 0), False],
+        ["rgba(255, 125, 33, 0.6)", Color(255, 125, 33, 0.6), False],
+        ["40% #405060", Color(64, 80, 96, 0.4), False],
+        ["hsl(45,25%,25%) 70%", Color(80, 72, 48, 0.7), False],
+        ["auto", Color(ANY, ANY, ANY, 1), True],
+        ["auto 25%", Color(ANY, ANY, ANY, 0.25), True],
+        ["32% auto", Color(ANY, ANY, ANY, 0.32), True],
+    ),
+)
+def test_parse_color_expression(
```

Comment:
Can we have one or two invalid color strings tests? i.e. the case where it fails.